tools/Google.Cloud.Tools.TagReleases/Program.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Octokit;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+using Newtonsoft.Json;
+
+using Google.Cloud.Tools.Common;
+
+namespace Google.Cloud.Tools.TagReleases
+{
+    /// <summary>
+    /// Tool to tag releases on GitHub for any projects where there's no existing tag
+    /// with the currently-specified version.
+    /// </summary>
+    internal class Program
+    {
+        private const string RepositoryOwner = "GoogleCloudPlatform";
+        private const string RepositoryName = "google-cloud-dotnet";
+        private const string ApplicationName = "google-cloud-dotnet-tagreleases";
+
+        private static int Main(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                Console.WriteLine("Arguments: <github access token>");
+                return 1;
+            }
+            try
+            {
+                MainAsync(args).GetAwaiter().GetResult();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine($"Error: {e}");
+                return 1;
+            }
+            return 0;
+        }
+
+        private static async Task MainAsync(string[] args)
+        {
+            var client = new GitHubClient(new ProductHeaderValue(ApplicationName))
+            {
+                Credentials = new Credentials(args[0])
+            };
+            var tags = (await client.Repository.GetAllTags(RepositoryOwner, RepositoryName)).Select(tag => tag.Name);
+
+            var apis = LoadApis();
```

Comment:
Mmm. I'm somewhat torn. I tend to regard quick tools like this as having a slightly lower bar than prod code (just like I use `var` more in test code) - not because it shouldn't be readable, just because the size of scope tends to be smaller. (It's also less likely that you *will* be reading tool code online, I suspect.)